vault/resource_transit_secret_backend_key.go

```diff
@@ -221,12 +235,13 @@ func transitSecretBackendKeyCreate(d *schema.ResourceData, meta interface{}) err
 		"derived":               d.Get("derived").(bool),
 		"type":                  d.Get("type").(string),
 		"auto_rotate_period":    autoRotatePeriod,
+		"key_size":              d.Get("key_size").(int),
```

Comment:
Looks like `key_size` was added in Vault 1.12. We should add a Vault Server version check in the resource and test files. You can see an examples of doing version checks in this PR: https://github.com/hashicorp/terraform-provider-vault/pull/1936/files#diff-3c100f49c020a28974d882dd5b34d276a7f825e3ccfaa18e584c1e26e2261d10R237

Reply:
I broke this out and set it separately from the block per your recommendation.